src/components/ViewStarship.vue

```diff
@@ -47,15 +54,46 @@ export default {
         starship_class: '',
         pilots: '',
       },
+      starshipImage: '',
+      starshipImageMsg: '',
+      imageShow: false,
+      loading: false,
     };
   },
   methods: {
     fetchRandomStarship() {
+      this.loading = true;
+      this.imageShow = false;
+      this.starshipImageMsg = '';
       const randomStarshipId = Math.floor((Math.random() * 37) + 1);
       starship.fetch(randomStarshipId).then((starshipData) => {
         this.starshipData = Object.assign(this.starshipData, starshipData);
+        starship.getPicture(this.starshipData.name).then((data) => {
+          this.loading = false;
+          if (data) {
+            this.starshipImage = data;
+            this.imageShow = true;
+          } else {
+            this.starshipImage = '';
+            this.starshipImageMsg = 'No image found.';
+          }
+        });
       });
     },
   },
 };
 </script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
+<style scoped>
```

Comment:
This block is copied across the components. Would be a good candidate to push to a style file that can be imported into components that need it.